src/indexer/index_writer.rs

```diff
@@ -167,20 +167,21 @@ pub(crate) fn advance_deletes(
     // TODO optimize
     // It should be possible to do something smarter by manipulation bitsets directly
     // to compute this union.
-    if let Some(seg_delete_bitset) = segment_reader.delete_bitset() {
+    if let Some(seg_alive_bitset) = segment_reader.alive_bitset() {
         for doc in 0u32..max_doc {
-            if seg_delete_bitset.is_deleted(doc) {
-                delete_bitset.insert(doc);
+            if seg_alive_bitset.is_deleted(doc) {
+                alive_bitset.remove(doc);
             }
         }
     }
 
-    let num_deleted_docs: u32 = delete_bitset.len() as u32;
+    let num_alive_docs: u32 = alive_bitset.num_set_bits() as u32;
+    let num_deleted_docs = max_doc - num_alive_docs;
```

Comment:
Actually there is `alive_bitset.num_deleted()`

Reply:
the alive_bitset is not the AliveBitSet, but just the im BitSet